test-lib.sh

```diff
@@ -1069,4 +1054,34 @@ ct_test_app_dockerfile() {
   return $ret
 }
 
+
+# ct_run_tests_from_testset
+# -----------------------------
+# Runs all tests in $TEST_SET, prints result to
+# the $TEST_SUMMARY variable
+# Argument: app_name - application name to log
+# Uses: $TEST_SET - set of test cases to run
+# Uses: $TEST_SUMMARY - variable for storing test results
+# Uses: $IMAGE_NAME - name of the image being tested
+ct_run_tests_from_testset() {
+  local app_name="$1"
+  local testsuite_result=0
```

Comment:
I would prefer to set it as a main variable at the beginning of each tests
```bash
TESTSUITE_RESULT=0
```bash

https://github.com/sclorg/s2i-perl-container/blob/master/test/run#L15

Reply:
If the variable is local, and its value is returned by this function, then the docker tests can use TESTSUITE_RESULT and openshift tests could use the OS_TESTSUITE_RESULT variable, as it is currently designed.

However, the truth is, that this would have to be implemented differently, for a case when the user calls SIGINT during the run of the test suite, because then [OS_]TESTSUITE_RESULT wouldn't be touched by the  `ct_run_tests_from_testset` function, and thus would return incoherent result. This could be fixed by setting the [OS_]TESTSUITE_RESULT to 1 in the beginning of the `run*` scripts.

Why do you prefer the global variable?

Reply:
What about if we run several ct_run_tests_from_testset. Not only the one. Then we have to if the function `ct_run_tests_from_testset` returns [0/1]. The global does this work.

Reply:
Well, this could be IMHO handled in the main `run*` script. It would be readable easier.